lochness/redcap/__init__.py

```diff
@@ -238,38 +238,40 @@ def get_run_sheets_for_datatypes(api_url, api_key,
 
             content_dict_list = json.loads(content)
 
-            content_num = 0
-            for content_dict in content_dict_list:
+            # for run sheet at each timepoint - baseline, follow up1, etc.
+            # content_num is set to start with 1 to match the session number
+            for content_num, content_dict in enumerate(content_dict_list, 1):
                 content_df = pd.DataFrame.from_dict(content_dict,
-                                                    orient='index')
+                                                    orient='index',
+                                                    columns=['field_value'])
+                content_df.index.name = 'field_name'
 
                 # if all is empty, or 0
                 all_empty = ((content_df[content_df.columns[0]]=='0') |
                              (content_df[content_df.columns[0]]=='')).all()
 
                 if all_empty:
                     continue  # don't have if all empty
-                else:
-                    content_num += 1
 
                 raw_modality_path = raw_path / modality
                 raw_modality_path.mkdir(exist_ok=True, parents=True)
                 output_name = Path(json_path).name.split('.json')[0]
                 
                 # output run sheet path
                 suffix = '' if content_num == 1 else f"_{content_num}"
-                if modality == 'surveys':
+                if modality == 'surveys':  # run sheet for PENN CNB
                     run_sheet_output = raw_modality_path / \
-                               f'{output_name}.Run_sheet_PennCNB{suffix}.csv'
+                       f'{output_name}.Run_sheet_PennCNB_{content_num}.csv'
 
                 else:
                     if run_sheet_name.endswith('checkin'):
                         run_sheet_output = raw_modality_path / \
                            f'{output_name}.Run_sheet_{modality}' \
-                           f'_checkin{suffix}.csv'
+                           f'_checkin_{content_num}.csv'
                     else:
                         run_sheet_output = raw_modality_path / \
-                           f'{output_name}.Run_sheet_{modality}{suffix}.csv'
+                           f'{output_name}.' \
+                           f'Run_sheet_{modality}_{content_num}.csv'
```

Comment:
The above run sheet naming scheme would require update here:
https://github.com/AMP-SCZ/utility/blob/d1f9a23c0ca2bd0a3424a573e1152939b340e1f8/subject_files_status_for_dpdash.py#L214-L215

Reply:
We can discuss if we should keep separate run sheets in the meeting today and decide if we want to keep this PR.